staging/src/k8s.io/apiserver/pkg/storage/cacher/watch_cache.go

```diff
@@ -257,6 +263,25 @@ func (w *watchCache) processEvent(event watch.Event, resourceVersion uint64, upd
 	return updateFunc(elem)
 }
 
+// Send a bookmark event with the current resourceVersion
+func (w *watchCache) sendBookmark() {
+	watchCacheEvent := &watchCacheEvent{
+		Type:   watch.Bookmark,
+		Object: w.newFunc(),
+	}
+
+	w.Lock()
```

Comment:
I thought about it in the context of reducing contention on this lock https://github.com/kubernetes/kubernetes/issues/73958#issuecomment-479522798
And I think that we can reduce it by changing the approach here.

What I would do is:
- move the sendBookmark logic fully to cacher
- sendBookmark doesn't set resourceVersion - it just triggers a bookmark event
- cacher is internally tracking last RV it processed
- when bookmark even is the one to procees, it uses that last processed RV to fill the object
- then we will be able to not call onEvent() under lock 

But I'm also fine with improving it in a separate PR.

Reply:
I will do it in a separate PR.

Reply:
Please add a TODO then.